units/en/unitbonus3/rl-for-trading.mdx

```diff
@@ -0,0 +1,162 @@
+
+### Example Workflow
+
+1. **Data Preparation:** Download historical stock data using libraries like Yahoo Finance API.
+
+```python
+from finrl.meta.preprocessor.yahoodownloader import YahooDownloader
+df = YahooDownloader(start_date='2009-01-01', end_date='2020-07-01', ticker_list=['AAPL', 'AMZN']).fetch_data()
+
+# Split df into df_train and df_test 
+```
+
+2. **Environment Configuration: **Create a trading environment using FinRL's StockTradingEnv class.
+
+```python
+from finrl.config import INDICATORS
+from finrl.meta.env_stock_trading.env_stocktrading import StockTradingEnv
+
+# This is only for illustration.
+# Values such as state_space, stock_dimension should be derived in the notebook before using them here.
+env_kwargs = {
+    "hmax": 100,
+    "initial_amount": 1000000,
+    "num_stock_shares": num_stock_shares,
+    "state_space": state_space,
+    "stock_dim": stock_dimension,
+    "tech_indicator_list": INDICATORS,
+    "action_space": stock_dimension,
+    "reward_scaling": 1e-4
+}
+
+env_train = StockTradingEnv(df=df_train, **env_kwargs)
+```
+
+3. **Agent Training: **Train a DRL agent using algorithms like DDPG or PPO from stable-baselines.
+
+There is also an option to train agent using ensemble strategy.
+
+```python
+from finrl.agents.stablebaselines3.models import DRLAgent
+agent = DRLAgent(env=env_train)
+agent.get_model('ppo')
+trained_ppo = agent.train_model()
+trained_ppo.save("trained_models/agent_ppo")
+```
+
+4. **Backtesting and Evaluation: **Evaluate the trained agent's performance using backtesting.
+
+Create a new environment using df_test to simulate test scenario.
+
+```python
+trained_ppo = PPO.load("trained_models/agent_ppo.zip")
+
+# This is only for illustration.
+# Values such as state_space, stock_dimension should be derived in the notebook before using them here.
+env_kwargs = {
+    "hmax": 100,
+    "initial_amount": 1000000,
+    "num_stock_shares": num_stock_shares,
+    "state_space": state_space,
+    "stock_dim": stock_dimension,
+    "tech_indicator_list": INDICATORS,
+    "action_space": stock_dimension,
+    "reward_scaling": 1e-4
+}
+
+env_test = StockTradingEnv(df=df_test, **env_kwargs)
+
+df_account_value_ppo, df_actions_ppo = DRLAgent.DRL_prediction(
+                                                model=trained_ppo, 
+                                                environment = e_test)
+```
+
+df_account_value_ppo, df_actions_ppo are further used for portfolio analysis.
+
+Their [tutorial series](https://finrl.readthedocs.io/en/latest/tutorial/Guide.html) is very exhaustive and beginner friendly. One can choose to start at various levels depending on familiarity (introduction, advance, practical, optimization, others).
+
+
+## Additional readings
```

Comment:
There is also this Gymnasium environment `gym-trading-env` that might be worth mentioning in this section (cf. https://gym-trading-env.readthedocs.io/, linked also on Gymnasium docs under the Third-party Environments section https://gymnasium.farama.org/environments/third_party_environments/#gym-trading-env-trading-environment).

Reply:
Thanks for your  suggestions, have included a link in additional readings.